### ISSUE TITLE
Ignore syntax errors caused by Liquid/Jekyll syntax

### ISSUE THREAD

**Issue by rugk** [eslint/eslint#6560]
> The version of ESLint you are using.

v2.13.1

> The problem you want to solve.

Basically I use [Jekyll](https://jekyllrb.com/) as a static site generator, which uses [Liquid](https://github.com/Shopify/liquid/wiki/Liquid-for-Designers) for manipulating files.
Using Liquid syntax in a JS file parsed by ESLint of course results in syntax-errors.
But e.g. this shows a very comfortable use case for Liquid syntax in JS files:

```js
window.myGlobalVar = {{ site.varApplyingForWholeProject }}
```

**So what happens is:** ESLint encounters syntax-error and *stops execution*(!). This means no actual syntax-checking happens.

Even disabling ESLint does not prevent this error:

```js
/* eslint-disable */

window.myGlobalVar = {{ site.varApplyingForWholeProject }}

/* eslint-enable */
```

> Your take on the correct solution to problem.

IMHO it would be nice if ESLint could at least ignore this error and proceed with syntax-checking. Possibly only if ESLint is disabled (via comments e.g.) or with a new "comment command" `eslint-disable-parsing`, which skips the whole part, which is marked as disabled.

## How other parsers are doing it

EsLint...
1. ignore lines, so that no errors are shown (there is only [an issue with ignoring a single-line](https://github.com/jshint/jshint/issues/2969))
2. continues after finding syntax errors

**Comment by michaelficarra** [comment]
This issue has been brought up a number of times before. See #2428, #900 and others. That is just not how eslint ignore comments work.

**Comment by btmills** [comment]
You could potentially build a [processor](http://eslint.org/docs/developer-guide/working-with-plugins#processors-in-plugins) that would support Liquid templates.

**Comment by rugk** [comment]
Or one with a `/* eslint-disable-parsing:start */` and `/* eslint-disable-parsing:end */`...
Or just integrate this into the default parser...

I mean if it was already requested many times I think it is certainly worth adding this.

**Comment by ilyavolodin** [comment]
@rugk ESLint is not a parser, it's a static code analysis tool. We use Espree (which is a layer on top of Acorn) as a parser. In order for ESLint to run, parser needs to be able to construct valid abstract syntax tree (AST). If you comment out part of the code and tell parser to skip over it, it might not be able to create valid AST.
As @btmills noted, you should create a processor to strip out all non-javascript syntax from the file, and pass only valid JS code to ESLint. People already built processors for HTML, Markdown and probably others, so you should be able to find some example and create one for Liquid templates.

**Comment by ilyavolodin** [comment]
As mentioned above, this is not something we can do. That's the reason processors were created.
Closing, as the question has been answered.
